### DISCUSSION TITLE
Is there something like onAfterMigrations()?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#4148] [Q&A]
I have a hook that syncs the pockethost.io user credentials `{ id, email, tokenKey, passwordHash }` to an `_admins` user in the pocketbase instance.

I am currently using `onAfterBootstrap`, but the very first time PocketBase runs it appears that this event is called before migrations are executed because I get `GoError: SQL logic error: no such table: _admins`.

After PocketBase runs once, everything is fine.

What hook should I be using to ensure migrations have run?

**Comment by ganigeorgiev** [comment]
There is no such hook at the moment.
As a hacky workaround you can use the `$app.onBeforeServe` but note that this is not concurrent safe and shouldn't be used for registering new routes with the JS pb_hooks. For an example you could check https://github.com/pocketbase/pocketbase/discussions/3220.

**Reply by benallfree** [reply]
I'm not able to get `$app.onBeforeServe` to fire. Do I need to put it inside an `onAfterBootstrap` or should it be at the top level? Neither seems to work.

**Reply by ganigeorgiev** [reply]
It should be at the top level.

**Reply by benallfree** [reply]
I see.

```js
$app.onBeforeServe().add( ()=>{
   // ...
})
```
